src/DataFrame/IO/Parquet/Binary.hs

```diff
@@ -30,10 +30,20 @@ word32ToLittleEndian w = map (\i -> fromIntegral (w `shiftR` i)) [0, 8, 16, 24]
 readUVarInt :: [Word8] -> (Word64, [Word8])
 readUVarInt xs = loop xs 0 0 0
   where
-    loop bs x _ 10 = (x, bs)
-    loop (b : bs) x s i
-        | b < 0x80 = (x .|. (fromIntegral b) `shiftL` s, bs)
-        | otherwise = loop bs (x .|. fromIntegral ((b .&. 0x7f) `shiftL` s)) (s + 7) (i + 1)
```

Comment:
There was a bug on this line: the original code would first run `shiftL` on a Word8 (potentially loosing bits by shifting "out of its 8 bits" and only then convert to Word64.
My code below makes sure we first convert to Word64 and only then shift.